# Notes for microbiota research 

Put daily work records on microbiota into this repository 



## Review

* [A practical guide to amplicon and metagenomic analysis of microbiome data](https://link.springer.com/article/10.1007/s13238-020-00724-8)
* [Current challenges and best-practice protocols for microbiome analysis](https://academic.oup.com/bib/article/22/1/178/5678919?login=false)
* [Shotgun metagenomics, from sampling to analysis](https://www.nature.com/articles/nbt.3935)
* [Targeting the gut and tumor microbiota in cancer](https://www.nature.com/articles/s41591-022-01779-2)
* [Microbiota in health and diseases](https://www.nature.com/articles/s41392-022-00974-4)
* [The gut microbiota–brain axis in behaviour and brain disorders](https://www.nature.com/articles/s41579-020-00460-0)
* [Gut microbiota in human metabolic health and disease](https://www.nature.com/articles/s41579-020-0433-9)
* [Gut microbiota and human NAFLD: disentangling microbial signatures from metabolic disorders](https://www.nature.com/articles/s41575-020-0269-9) 
* [Microbiome datasets are compositional: And this is not optional](https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full)
* [Critical Assessment of Metagenome Interpretation: the second round of challenges](https://www.nature.com/articles/s41592-022-01431-4)
* [Next steps after 15 stimulating years of human gut microbiome research](https://sfamjournals.onlinelibrary.wiley.com/doi/full/10.1111/1751-7915.13970)
* [Rethinking healthy eating in light of the gut microbiome](https://www.cell.com/cell-host-microbe/fulltext/S1931-3128(22)00222-0)
* [Ecosystem-specific microbiota and microbiome databases in the era of big data](https://environmentalmicrobiome.biomedcentral.com/articles/10.1186/s40793-022-00433-1)



## Method

* [Analysis of microbial compositions: a review of normalization and differential abundance analysis](https://www.nature.com/articles/s41522-020-00160-w)
* [Microbiome differential abundance methods produce different results across 38 datasets](https://www.nature.com/articles/s41467-022-28034-z)



## Research article

### Longitudinal study

* [Acute and persistent effects of commonly used antibiotics on the gut microbiome and resistome in healthy adults](https://www.sciencedirect.com/science/article/pii/S2211124722004016#:~:text=In%20conclusion%2C%20our%20findings%20indicate,in%20AR%20in%20healthy%20microbiomes.)

### NAFLD: non-alcoholic fatty liver disease

* [Gut microbiota and human NAFLD: disentangling microbial signatures from metabolic disorders](https://www.nature.com/articles/s41575-020-0269-9)
* [Characterization of microbiome and metabolite analyses in patients with metabolic associated fatty liver disease and type II diabetes mellitus](https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-022-02526-w)
* [Multiomics Analysis Reveals the Impact of Microbiota on Host Metabolism in Hepatic Steatosis](https://onlinelibrary.wiley.com/doi/10.1002/advs.202104373)


### Mechanism

* [Microbiota alterations in proline metabolism impact depression](https://www.cell.com/cell-metabolism/fulltext/S1550-4131(22)00128-0?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1550413122001280%3Fshowall%3Dtrue)


### ICI 

+ [Intestinal Akkermansia muciniphila predicts clinical response to PD-1 blockade in patients with advanced non-small-cell lung cancer](https://www.nature.com/articles/s41591-021-01655-5)



### Blood microbiota

* [﻿Microbiome analyses of blood and tissues suggest cancer diagnostic approach](https://www.nature.com/articles/s41586-020-2095-1)






## Database

* [Integrated Microbial Genomes and Microbiomes (JGI-IMG/M)](https://img.jgi.doe.gov/)
* [16S_rRNA_Microbiome_Datasets](https://figshare.com/articles/dataset/16S_rRNA_Microbiome_Datasets/14531724)
* [Earth Microbiome Project (EMP)](https://earthmicrobiome.org/)
* [Human Oral Microbiome Database (HOMD)](https://www.homd.org/)
* [Integrative Human Microbiome Project (HMP)](https://commonfund.nih.gov/hmp/databases)
* [Unifed Human Gastrointestinal Genome (UHGG)](https://www.ebi.ac.uk/metagenomics/genome-catalogues/human-gut-v2-0)
* [Global Microbial Gene Catalog (GMGC)](https://gmgc.embl.de/)
* [The Comprehensive Antibiotic Resistance Database](https://card.mcmaster.ca/)
* [microbioTA: an atlas of the microbiome in multiple disease tissues of Homo sapiens and Mus musculus](http://bio-annotation.cn/microbiota/)



## Benchmark

* [Comprehensive benchmarking and ensemble approaches for metagenomic classifiers](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1299-7?gclid=Cj0KCQjwnNyUBhCZARIsAI9AYlHQvfX8NY5nSog986R76_fs-HJvZaBpf9eTRRN7vy-UZyJwkfDFbbsaAtRvEALw_wcB)



## Pipeline

* Humann2: [Species-level functional profiling of metagenomes and metatranscriptomes](https://www.nature.com/articles/s41592-018-0176-y)



## Data Analysis Materials

* [microViz](https://david-barnett.github.io/microViz/index.html)
* [mixOmics](https://mixomicsteam.github.io/Bookdown/intro.html)
* [Advanced Regression Methods](https://bookdown.org/chua/ber642_advanced_regression/)
* [Orchestrating Microbiome Analysis](https://microbiome.github.io/OMA/)
* [Advanced Multivariate Analyses in R](http://r.qcbs.ca/workshop10/book-en/index.html#preface)
* [Statistical tools for high-throughput data analysis: STHDA](http://www.sthda.com/english/)
* [Statistical Analysis of Microbiome Data with R](https://link.springer.com/book/10.1007/978-981-13-1534-3)
* [Doing Meta-Analysis with R: A Hands-On Guide](https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/)
* [Constrained ordination](https://fukamilab.github.io/BIO202/06-B-constrained-ordination.html)
* [MACHINE LEARNING 2021 SPRING](https://speech.ee.ntu.edu.tw/~hylee/ml/2021-spring.php)
* [Visualisation of proteomics data using R and Bioconductor](https://lgatto.github.io/RforProteomics/articles/RProtVis.html)
* [LightGBM R-package](https://lightgbm.readthedocs.io/en/latest/R/index.html)
* [Mastering Software Development in R](https://bookdown.org/rdpeng/RProgDA/)
  



## Lab

* [Zeller Team](https://www.embl.org/groups/zeller/): The Zeller team develops analysis strategies and tools to investigate how the microbiome contributes to human health, disease progression and treatment success, and how it is shaped by host factors such as nutrition and drug intake.
* [Huttenhower lab](https://huttenhower.sph.harvard.edu/): The lab try to learn to use the microbiome to predict disease onset, progression, and outcomes, as well as developing ways to modify it for health maintenance and therapy.
* [KOST LAB](https://www.kostlab.com/christian-kost.html): My working group uses tractable, laboratory-based model systems to address fundamental biological questions on different levels of biological organization.
* [Daniel Croll lab](https://pathogen-genomics.org/): The lab of Daniel Croll wants to understand how pathogens evolve to cause disease.
* [Honda Lab](http://www.microbiolimmunol.med.keio.ac.jp/): We aim to identify specific intestinal bacterial species that influence the host immune cells by inducing their functions and differentiation.
* [Borenstein lab](http://borensteinlab.com/): computational study of the human microbiome and of other complex microbial ecosystems.
* [Gore Lab](http://www.gorelab.org/index.html): The Gore Lab studies how interactions between individuals determine the evolutionary and ecological dynamics of multi-species microbial communities. Of particular focus are alternative stables states, community assembly, cross-feeding, and the emergence of "cheater" strategies. The laboratory is composed of an interdisciplinary group of scientists interested in learning from each  other to effectively combine experiments, theory, and modeling.
  



## Job view

* [Science Jobs hunter](https://jobrxiv.org/)
