---
title: "Omics DataSet from Zeybel 2022"
date: "Created: 2022-12-07; Updated: `r Sys.Date()`"
author:
  - name: "Hua Zou"
    email: "zouhua1@outlook.com"
output:
  html_notebook:
    # codes: hide  
    # html_document: 
    #     toc: true
---

## Introduction

Generating datasets which are phyloseq-class object for microbiota and SummarizedExperiment-class object for metabolites from 

Zeybel et al 2022 paper.


+ **phyloseq**: phyloseq-class object

  - otu table
  
  - taxa table
  
  - sample table
  
  - tree file

  - Representative sequences (ASV OTU)

+ **SummarizedExperiment**: SummarizedExperiment-class object

  - otu table
  
  - taxa table
  
  - sample table

```{r setup}
knitr::opts_chunk$set(message = FALSE, warning = FALSE)
library(dplyr)
library(tibble)
library(phyloseq)
library(SummarizedExperiment)

# rm(list = ls())
options(stringsAsFactors = F)
options(future.globals.maxSize = 1000 * 1024^2)
```


## Importing Data


### cross-section data


56 participants


+ metadata

  - Patients' Phenotypic information (sheet 2)
  
  - Clinical and Physical variables (sheet 3) 

+ taxa of fecal samples 

  - Metaphlan2 profile of Gut microbiota (sheet 4) 
  
  - Metaphlan2 profile of Oral microbiota (sheet 5) 

+ metabolites of fecal samples 

  - Metabolomics of Patients' plasma samples (sheet 6) 

+ proteomics of host plasma

  - Inflammatory Proteomics of Patients' plasma samples (sheet 7) 


```{r}
meta_info <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 2)
meta_clin <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 3)

taxa_gut <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 4)
taxa_ora <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 5)

metabolites_fecal <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 6)

protein_plasma <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 7)
```


### longitudinal data

22 of these patients after 2â€“3 months

+ metadata

  - Patients' Phenotypic information (sheet 8) 
  
  - Clinical and Physical variables (sheet 9) 

+ taxa of fecal samples 

  - Metaphlan2 profile of Gut microbiota (sheet 10) 
  
  - Metaphlan2 profile of Oral microbiota (sheet 11) 

+ metabolites of fecal samples 

  - Metabolomics of Patients' plasma samples (sheet 12) 

+ proteomics of host plasma

  - Inflammatory Proteomics of Patients' plasma samples (sheet 13) 


```{r}
long_meta_info <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 8)
long_meta_clin <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 9)

long_taxa_gut <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 10)
long_taxa_ora <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 11)

long_metabolites_fecal <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 12)

long_protein_plasma <- readxl::read_xlsx("./OmicsDataSet-Zeybel et al. - 2022.xlsx", sheet = 13)
```



## cross-section data

56 participants

+ metadata
```{r}
metadata <- meta_info %>%
  dplyr::select(-`Patient ID`) %>%
  # dplyr::rename(`Patient ID` = `Patient ID...2`) %>%
  dplyr::inner_join(meta_clin,
                    by = "PatientID") %>%
  dplyr::select(-c("Stage", "Metabolomics", "Proteomics", 
                   "GutMetagenomics", "OralMetagenomics")) %>%
  dplyr::select(PatientID, Gender, Age, everything())

head(metadata)
```

+ phyloseq-class object
```{r}
import_metaphlan_taxa <- function(data_metaphlan2,
                                  taxa_level = NULL) {

  taxa_rank <- c("Kingdom", "Phylum", "Class",
                 "Order", "Family", "Genus",
                 "Species", "Strain")

  # rename 1st column into "ID"
  colnames(data_metaphlan2)[which(colnames(data_metaphlan2) == "ID" |
                                    colnames(data_metaphlan2) == "clade_name")] <- "ID"

  # remove the "NCBI_tax_id" column
  if (any(colnames(data_metaphlan2) %in% "NCBI_tax_id")) {
    data_metaphlan2 <- data_metaphlan2 %>% 
      dplyr::select(-NCBI_tax_id)
  }

  # remove sample names with "metaphlan_bugs_list" suffix
  colnames(data_metaphlan2) <- gsub("_metaphlan_bugs_list$", "", 
                                    colnames(data_metaphlan2))

  if (is.null(taxa_level)) {
    taxa_level <- "Species"
  }
  ind_number <- match(taxa_level, taxa_rank)

  # remove k__Archaea & k__Viruses & k__Eukaryota & Others which are not bacteria
  if (length(grep("k__Bacteria", data_metaphlan2$ID)) > 0) {
    taxa_bacteria <- data_metaphlan2 %>%
      dplyr::slice(grep("k__Bacteria", ID)) %>%
      tibble::column_to_rownames("ID")
  } else {
    stop("There are no bacteria identified in this profile")
  }

  # dividing by 100 to normalize the sum into 1
  taxa_bac_per <- (taxa_bacteria / 100)  %>%
    tibble::rownames_to_column("ID")

  taxa <- taxa_bac_per %>%
    dplyr::group_by(ID) %>%
    dplyr::mutate(Number=length(unlist(strsplit(ID, "|", fixed = TRUE)))) %>%
    dplyr::select(ID, Number) %>%
    dplyr::filter(Number == ind_number)

  abundance_table <- taxa_bac_per %>%
    dplyr::filter(ID%in%taxa$ID)

  taxa_table <- lapply(abundance_table$ID, strsplit, split = "|", fixed = TRUE)
  taxa_table <- lapply(taxa_table, unlist)
  taxa_table <- do.call(rbind, taxa_table)
  taxa_table <- data.frame(taxa_table)
  names(taxa_table) <- taxa_rank[1:ind_number]
  rownames(taxa_table) <- taxa_table[, ind_number]

  abundance_table_res <- abundance_table %>%
    tibble::column_to_rownames("ID")
  rownames(abundance_table_res) <- rownames(taxa_table)
  abundance_table_res <- abundance_table_res[match(rownames(taxa_table),
                                                   row.names(abundance_table_res)), ,
                                             drop = FALSE]

  res <- list(tax_tab=taxa_table,
              abu_tab=abundance_table_res)

  return(res)
}

get_metaphlan_phyloseq <- function(otu_tab,
                                   sam_tab,
                                   tax_tab = NULL) {

  sid <- intersect(sam_tab$PatientID, colnames(otu_tab))
  
  sam_tab <- sam_tab[sam_tab$PatientID %in% sid, ] %>%
    tibble::column_to_rownames("PatientID")
  otu_tab <- otu_tab %>%
    dplyr::select(all_of(rownames(sam_tab)))
  
  
  if (!is.null(tax_tab)) {
    tax_tab <- phyloseq::tax_table(as(tax_tab, "matrix"))
  }

  if (!is.null(sam_tab)) {
    sam_tab <- phyloseq::sample_data(sam_tab %>% data.frame())
  }

  asv_tab <- phyloseq::otu_table(as(otu_tab, "matrix"),
                                 taxa_are_rows = TRUE)

  ps <- phyloseq::phyloseq(asv_tab, tax_tab, sam_tab)

  return(ps)
}


taxa_gut_list <- import_metaphlan_taxa(data_metaphlan2 = taxa_gut)
taxa_ora_list <- import_metaphlan_taxa(data_metaphlan2 = taxa_ora)

ps_gut <- get_metaphlan_phyloseq(otu_tab = taxa_gut_list$abu_tab,
                                 sam_tab = metadata,
                                 tax_tab = taxa_ora_list$tax_tab)
ps_gut

ps_ora <- get_metaphlan_phyloseq(otu_tab = taxa_ora_list$abu_tab,
                                 sam_tab = metadata,
                                 tax_tab = taxa_ora_list$tax_tab)
ps_ora
```


+ SummarizedExperiment: metabolites
```{r}
metabolites_fecal$metabolitesID <- paste0("Chem_", metabolites_fecal$CHEMICALID)

# Row->metabolites; Col->Samples
metabolites_fecal_assay <- metabolites_fecal %>%
  dplyr::select(c("metabolitesID", c(starts_with("P10")))) %>%
  tibble::column_to_rownames("metabolitesID")

metabolites_fecal_rowData <- metabolites_fecal %>%
  dplyr::select(-c(starts_with("P10"))) %>%
  dplyr::select(metabolitesID, everything())

metabolites_fecal_colData <- metadata  %>%
  dplyr::filter(PatientID %in% all_of(colnames(metabolites_fecal_assay)))

se_metabolite <- SummarizedExperiment(
                     assays = metabolites_fecal_assay,
                     rowData = metabolites_fecal_rowData,
                     colData = metabolites_fecal_colData,
                     checkDimnames = TRUE)

se_metabolite
```

+ SummarizedExperiment: proteomics
```{r}
protein_plasma_new <- protein_plasma %>%
  dplyr::rename(ProteinID=SampleName)
protein_plasma_new$ProteinID <- gsub("-", "_", protein_plasma_new$ProteinID)

# Row->proteins; Col->Samples
protein_plasma_assay <- protein_plasma_new %>%
  dplyr::select(c("ProteinID", c(starts_with("P10")))) %>%
  tibble::column_to_rownames("ProteinID")

protein_plasma_rowData <- protein_plasma_new %>%
  dplyr::select(-c(starts_with("P10"))) %>%
  dplyr::select(ProteinID, everything())

protein_plasma_colData <- metadata %>%
  dplyr::filter(PatientID %in% all_of(colnames(protein_plasma_assay)))

se_protein <- SummarizedExperiment(
                     assays = protein_plasma_assay,
                     rowData = protein_plasma_rowData,
                     colData = protein_plasma_colData,
                     checkDimnames = TRUE)

se_protein
```


+ output
```{r}
if (!dir.exists("./result/")) {
  dir.create("./result/", recursive = TRUE)
}

saveRDS(ps_gut, "./result/Zeybel_2022_gut_MGS_ps.RDS", compress = TRUE)
saveRDS(ps_ora, "./result/Zeybel_2022_oral_MGS_ps.RDS", compress = TRUE)
saveRDS(se_metabolite, "./result/Zeybel_2022_fecal_metabolite_se.RDS", compress = TRUE)
saveRDS(se_protein, "./result/Zeybel_2022_plasma_protein_se.RDS", compress = TRUE)
```



## longitudinal data

+ metadata
```{r}
# after
metadata_after <- long_meta_info %>%
  dplyr::inner_join(long_meta_clin,
                    by = "PatientID") %>%
  dplyr::select(PatientID, Gender, Age, everything())


# before
metadata_before <- metadata %>%
  dplyr::filter(PatientID %in% metadata_after$PatientID) %>%
  dplyr::mutate(Stage = "Before")

# cbind
long_metadata <- rbind(metadata_before, metadata_after) %>%
  dplyr::mutate(SampleID = paste(PatientID, Stage, sep = "_")) %>%
  dplyr::select(SampleID, PatientID, Gender, Age, Stage, 
                LiverFatClass, 
                Weight, Body_mass_index, 
                Hip_circumference, 
                Waist_circumference, everything()) %>%
  dplyr::arrange(PatientID, desc(Stage))

head(long_metadata)
```

+ phyloseq-class object
```{r}
get_long_taxa <- function(x, y) {
  
  # x = long_taxa_gut
  # y = taxa_gut_list
  
  dat <- x
  otu_tab <- y$abu_tab
  tax_tab <- y$tax_tab  
  
  dat$TaxaID <- paste0("s__", dat$TaxaID)
  dat <- dat %>%
    tibble::column_to_rownames("TaxaID")
  dat_cln <- dat[, !is.na(colSums(dat))]
  sid <- intersect(colnames(dat_cln), colnames(otu_tab))
  # before
  dat_before <- otu_tab %>%
    dplyr::select(all_of(sid))
  dat_after <- dat_cln %>%
    dplyr::select(all_of(sid))
  
  colnames(dat_before) <- paste0(colnames(dat_before), "_Before")  
  colnames(dat_after) <- paste0(colnames(dat_after), "_After")  
  
  mdat <- dat_before %>%
    tibble::rownames_to_column("TaxaID") %>%
    dplyr::inner_join(dat_after %>%
                        tibble::rownames_to_column("TaxaID"),
                      by = "TaxaID") %>%
    tibble::column_to_rownames("TaxaID")
  
  tax_tab_mdat <- tax_tab[rownames(tax_tab)%in%rownames(mdat), , ]
  
  
  res <- list(abu_tab = mdat,
              tax_tab = tax_tab_mdat)
  
  return(res)
}


get_metaphlan_phyloseq2 <- function(otu_tab,
                                   sam_tab,
                                   tax_tab = NULL) {

  sid <- intersect(sam_tab$SampleID, colnames(otu_tab))
  
  sam_tab <- sam_tab[sam_tab$SampleID %in% sid, ] %>%
    tibble::column_to_rownames("SampleID")
  otu_tab <- otu_tab %>%
    dplyr::select(all_of(rownames(sam_tab)))
  
  
  if (!is.null(tax_tab)) {
    tax_tab <- phyloseq::tax_table(as(tax_tab, "matrix"))
  }

  if (!is.null(sam_tab)) {
    sam_tab <- phyloseq::sample_data(sam_tab %>% data.frame())
  }

  asv_tab <- phyloseq::otu_table(as(otu_tab, "matrix"),
                                 taxa_are_rows = TRUE)

  ps <- phyloseq::phyloseq(asv_tab, tax_tab, sam_tab)

  return(ps)
}

long_taxa_gut_list <- get_long_taxa(x = long_taxa_gut, y = taxa_gut_list)
long_taxa_ora_list <- get_long_taxa(x = long_taxa_ora, y = taxa_ora_list)

long_ps_gut <- get_metaphlan_phyloseq2(
                      otu_tab = long_taxa_gut_list$abu_tab,
                      sam_tab = long_metadata,
                      tax_tab = long_taxa_gut_list$tax_tab)
long_ps_gut

long_ps_ora <- get_metaphlan_phyloseq2(
                      otu_tab = long_taxa_ora_list$abu_tab,
                      sam_tab = long_metadata,
                      tax_tab = long_taxa_ora_list$tax_tab)
long_ps_ora
```

+ SummarizedExperiment: metabolites
```{r}
metabolites_ID <- intersect(metabolites_fecal$BIOCHEMICAL, long_metabolites_fecal$MetaboliteID)
long_metabolites_fecal_rowData <- metabolites_fecal_rowData %>%
  dplyr::filter(BIOCHEMICAL %in% metabolites_ID)

# Row->metabolites; Col->Samples
long_metabolites_fecal_assay_after <- long_metabolites_fecal %>%
  dplyr::inner_join(long_metabolites_fecal_rowData %>% 
                      dplyr::select(metabolitesID, BIOCHEMICAL),
                    by = c(MetaboliteID = "BIOCHEMICAL")) %>%
  dplyr::select(-MetaboliteID) %>%
  dplyr::select(c("metabolitesID", c(starts_with("P10")))) %>%
  tibble::column_to_rownames("metabolitesID")

meta_sid <- intersect(colnames(long_metabolites_fecal_assay_after),
                 colnames(metabolites_fecal_assay))
long_metabolites_fecal_assay_after <- long_metabolites_fecal_assay_after %>%
  dplyr::select(all_of(meta_sid))

long_metabolites_fecal_assay_before <- 
  metabolites_fecal_assay[rownames(metabolites_fecal_assay) %in% rownames(long_metabolites_fecal_assay_after),
                          colnames(metabolites_fecal_assay) %in% colnames(long_metabolites_fecal_assay_after)]

colnames(long_metabolites_fecal_assay_before) <- paste0(colnames(long_metabolites_fecal_assay_before), "_Before")  
colnames(long_metabolites_fecal_assay_after) <- paste0(colnames(long_metabolites_fecal_assay_after), "_After") 

long_metabolites_fecal_assay <- long_metabolites_fecal_assay_before %>%
  tibble::rownames_to_column("TempRowNames") %>%
  dplyr::inner_join(long_metabolites_fecal_assay_after %>%
                      tibble::rownames_to_column("TempRowNames"),
                    by = "TempRowNames") %>%
  tibble::column_to_rownames("TempRowNames")

long_metabolites_fecal_colData <- long_metadata  %>%
  dplyr::filter(SampleID %in% all_of(colnames(long_metabolites_fecal_assay))) 

long_se_metabolite <- SummarizedExperiment(
                     assays = long_metabolites_fecal_assay,
                     rowData = long_metabolites_fecal_rowData,
                     colData = long_metabolites_fecal_colData,
                     checkDimnames = TRUE)

long_se_metabolite
```

+ SummarizedExperiment: proteomics
```{r}
long_protein_plasma$ProteinID <- gsub("-", "_", long_protein_plasma$ProteinID)
proteins_ID <- intersect(protein_plasma_new$ProteinID, long_protein_plasma$ProteinID)


# Row->proteins; Col->Samples
long_protein_plasma_assay_after <- long_protein_plasma %>%
  dplyr::filter(ProteinID %in% proteins_ID) %>%
  dplyr::select(c(ProteinID, c(starts_with("P10")))) %>%
  tibble::column_to_rownames("ProteinID")

pron_id <- intersect(colnames(long_protein_plasma_assay_after),
                     colnames(protein_plasma_assay))
long_protein_plasma_assay_before <- protein_plasma_assay %>%
  dplyr::select(all_of(pron_id))
long_protein_plasma_assay_after <- long_protein_plasma_assay_after %>%
  dplyr::select(all_of(pron_id))

colnames(long_protein_plasma_assay_before) <- paste0(colnames(long_protein_plasma_assay_before), "_Before")  
colnames(long_protein_plasma_assay_after) <- paste0(colnames(long_protein_plasma_assay_after), "_After") 

long_protein_plasma_assay <- long_protein_plasma_assay_before %>%
  tibble::rownames_to_column("TempRowNames") %>%
  dplyr::inner_join(long_protein_plasma_assay_after %>%
                      tibble::rownames_to_column("TempRowNames"),
                    by = "TempRowNames") %>%
  tibble::column_to_rownames("TempRowNames")

long_protein_plasma_rowData <- protein_plasma_rowData %>%
  dplyr::rename(`Protein ID` = ProteinID)

long_protein_plasma_colData <- long_metadata  %>%
  dplyr::filter(SampleID %in% all_of(colnames(long_protein_plasma_assay))) 

long_se_protein <- SummarizedExperiment(
                     assays = long_protein_plasma_assay,
                     rowData = long_protein_plasma_rowData,
                     colData = long_protein_plasma_colData,
                     checkDimnames = TRUE)

long_se_protein
```


+ output
```{r}
if (!dir.exists("./result/")) {
  dir.create("./result/", recursive = TRUE)
}

saveRDS(long_ps_gut, "./result/Zeybel_2022_gut_MGS_ps_Paired.RDS", compress = TRUE)
saveRDS(long_ps_ora, "./result/Zeybel_2022_oral_MGS_ps_Paired.RDS", compress = TRUE)
saveRDS(long_se_metabolite, "./result/Zeybel_2022_fecal_metabolite_se_Paired.RDS", compress = TRUE)
saveRDS(long_se_protein, "./result/Zeybel_2022_plasma_protein_se_Paired.RDS", compress = TRUE)
```


## Session info
```{r}
devtools::session_info()
```


## Reference

+ [Multiomics Analysis Reveals the Impact of Microbiota on Host Metabolism in Hepatic Steatosis](https://onlinelibrary.wiley.com/doi/full/10.1002/advs.202104373)

+ [Basic storage, access, and manipulation of phylogenetic sequencing data with phyloseq](https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-basics.html)

+ [Practical: Organizing data with SummarizedExperiment](https://bioconductor.org/help/course-materials/2019/BSS2019/04_Practical_CoreApproachesInBioconductor.html)
